### PR TITLE
PAINTROID-265 Refactor TextTool to Kotlin

### PR DESCRIPTION
**Pull request by CodeChamp-SS**
Files that were refactored:

1. TextTool
2. TextToolOptionsView

https://jira.catrob.at/browse/PAINTROID-265

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Review by geolas94 — CHANGES_REQUESTED** [review]

**Review by geolas94 — APPROVED** [review]

**Review by geolas94 — APPROVED** [review]
lgtm

**Comment by ThorstenBandel** [comment]
There are static analysis errors and warnings, could you please fix them ? :)

**Comment by CodeChamp-SS** [comment]
> There are static analysis errors and warnings, could you please fix them ? :)

sure

File: Paintroid/src/main/java/org/catrobat/paintroid/tools/options/TextToolOptionsView.kt [hunk 1]
```diff
@@ -16,27 +16,18 @@
...
-
-		void hideToolOptions();
-	}
+package org.catrobat.paintroid.tools.options
+
+interface TextToolOptionsView {
+    fun setState(bold: Boolean, italic: Boolean, underlined: Boolean, text: String, textSize: Int, font: String)
+    fun setCallback(listener: Callback)
+    interface Callback {
+        fun setText(text: String)
```
[the remark this answers is not in the document]
Comment (CodeChamp-SS) [reply Paintroid/src/main/java/org/catrobat/paintroid/tools/options/TextToolOptionsView.kt:34]: added the empty lines :)

**Comment by wslany** [comment]
Pls could you once more resolve the conflicts?

**Review by ThorstenBandel — APPROVED** [review]
Lgtm :D

**Pull request merged**